tests/integration/targets/ec2_spot_instance/tasks/main.yaml

```diff
@@ -145,6 +157,35 @@
       launch_spec: '{{ complex_create_result.spot_request.launch_specification }}'
       spot_request_tags: '{{ complex_create_result.spot_request.tags }}'
 
+  - name: Get info about the complex spot instance request created
+    ec2_spot_instance_info:
+      spot_instance_request_ids:
+        - "{{ complex_create_result.spot_request.spot_instance_request_id }}"
+    register: complex_spot_instance_info_result
+
+  - name: Assert that the complex spot request created is open/active and correct keys are set
+    assert:
+      that:
+        - complex_spot_instance_info_result.spot_request[0].state in ['open', 'active']
+        - complex_create_result.spot_request.spot_price == "0.002000"
+        - complex_create_result.spot_request.launch_specification.block_device_mappings[0].ebs.volume_size == 5
+        - complex_create_result.spot_request.launch_specification.block_device_mappings[0].ebs.volume_type == "gp3"
+        - complex_create_result.spot_request.launch_specification.network_interfaces[0].delete_on_termination == True
```

Comment:
For these we want to compare the info_result to the create_result to ensure the two modules have consistent results. So these would be:
```suggestion
        - complex_create_info_result.spot_request.spot_price == complex_create_result.spot_request.spot_price
        - complex_create_info_result.spot_request.launch_specification.block_device_mappings[0].ebs.volume_size == complex_create_result.spot_request.launch_specification.block_device_mappings[0].ebs.volume_size
        - complex_create_info_result.spot_request.launch_specification.block_device_mappings[0].ebs.volume_type == complex_create_result.spot_request.launch_specification.block_device_mappings[0].ebs.volume_type
        - complex_create_info_result.spot_request.launch_specification.network_interfaces[0].delete_on_termination == complex_create_result.spot_request.launch_specification.network_interfaces[0].delete_on_termination
```

In this specific case though, that gets really long and hard to read. So I would suggest using a task var for all of the launch spec assertions like the example at line 156-158.  Which could make this block instead be something like:

```suggestion
        - complex_create_info_result.spot_request.spot_price == complex_create_result.spot_request.spot_price 
        - info_launch_spec.block_device_mappings[0].ebs.volume_size == create_launch_spec.block_device_mappings[0].ebs.volume_size
        - info_launch_spec.block_device_mappings[0].ebs.volume_type == create_launch_spec.block_device_mappings[0].ebs.volume_type
        - info_launch_spec.network_interfaces[0].delete_on_termination == create_launch_spec.network_interfaces[0].delete_on_termination
    vars:
      create_launch_spec: "{{ complex_create_result.spot_request.launch_specification }}"
      info_launch_spec: "{{ complex_create_info_result.spot_request.launch_specification }}"

```
(I'm reasonably confident my indentation is correct on that one, but it's hard to tell for sure in the github comment pop-up.  :))